pkg/controllers/managementuser/clusterauthtoken/cluster_user_attribute.go

```diff
@@ -17,15 +23,32 @@ func (h *clusterUserAttributeHandler) Sync(key string, clusterUserAttribute *clu
 		return nil, nil
 	}
 
-	if !clusterUserAttribute.NeedsRefresh {
-		return nil, nil
-	}
-
 	userAttribute, err := h.userAttributeLister.Get("", clusterUserAttribute.Name)
 	if err != nil {
-		return nil, err
+		// If the corresponding UserAttribute no longer exists, we need to clean up the ClusterUserAttribute.
+		if apierrors.IsNotFound(err) {
+			// There is a chance that it hasn't ended up in the cache yet so we try to get it afresh.
```

Comment:
I think this could be simplified to:

```go
if !apierrors.IsNotFound(err) {
    return nil, err
}
// There is a chance that it hasn't ended up in the cache yet so we try to get it afresh.
userAttribute, err = h.userAttribute.Get(clusterUserAttribute.Name, metav1.GetOptions{})
```

Per the Go Review guidelines... https://go.dev/wiki/CodeReviewComments#indent-error-flow

Reply:
It's not exactly the same case as described in the guidelines, but yes we can minimize the indentation twice here. Though we'll need to forward-port it back to 2.9 where this originated from.